be/src/util/threadpool.h

```diff
@@ -362,6 +362,13 @@ class ThreadPoolToken {
     // Returns true if all submissions are complete, false otherwise.
     bool wait_for(const MonoDelta& delta);
 
+    bool need_dispatch();
```

Comment:
need_dispatch is a bit weird. Idle may be better or have other better names

Reply:
The purpose of this `need_dispatch` is to express whether the token needs to be put into the queue, so it does not actually mean idle.